stats.js

```diff
@@ -188,6 +185,26 @@ config.configFile(process.argv[2], function (config) {
 
   l = new logger.Logger(config.log || {});
 
+  // force conf.gaugesMaxTTL to 1 if it not a positive integer > 1
+  if (helpers.isInteger(conf.gaugesMaxTTL) && conf.gaugesMaxTTL > 1) {
+    conf.gaugesMaxTTL = conf.gaugesMaxTTL;
+  } else {
+    conf.gaugesMaxTTL = 1;
+  }
+
+  // allows us to flag all of these on with a single config but still override them individually
+  conf.deleteIdleStats = conf.deleteIdleStats !== undefined ? conf.deleteIdleStats : false;
+  if (conf.deleteIdleStats) {
+    conf.deleteCounters = conf.deleteCounters !== undefined ? conf.deleteCounters : true;
+    conf.deleteTimers = conf.deleteTimers !== undefined ? conf.deleteTimers : true;
+    conf.deleteSets = conf.deleteSets !== undefined ? conf.deleteSets : true;
+    conf.deleteGauges = conf.deleteGauges !== undefined ? conf.deleteGauges : true;
+  }
+
+  // if gauges are not being deleted, clear gaugesTTL counters to save memory
+  if (! conf.deleteGauges) {
```

Comment:
Nice optimisation and thanks for the comment, this is super simple and very clear, I like it :smile: